File: Lectures/03-Interacting-Objects/TimeUnit.java
public class TimeUnit {
    private int myLimit;
    private int myValue;
    private String myUnit;

    public TimeUnit() { this(60); }

    public TimeUnit(int theLimit) { this(theLimit, ""); }

    public TimeUnit(int theLimit, String theUnit) {
      myLimit = theLimit;
      myUnit = theUnit;
      myValue = 0;
    }

    public boolean update() {
      myValue = ++myValue % myLimit;
      return (0 == myValue);
    }

    public String getDisplayValue() {
      return String.format("%02d", myValue);
/*      if (10 > myvalue) {
        return "0" + myValue;
      } else {
        return "" + myValue;
      }*/
    }
  }

File: Lectures/03-Interacting-Objects/timeunit.cc
#include <string>
using namespace std;

#include "timeunit.hh"

TimeUnit::TimeUnit(int theLimit, const char* theUnit) : myLimit(theLimit),myUnit(theUnit) {
  myValue = 0;
}

bool TimeUnit::update(void) {
  myValue = ++myValue % myLimit;
  return (0 == myValue);
}

string TimeUnit::getDisplayValue(void) { // There is the c function sprintf() which we could use similarly to what
  if (myValue<10) {                      // we did for java, but it is more of a bother.
    return string("0")+std::to_string(myValue);
  } else {
    return std::to_string(myValue);
  }
}
